demos/chat_with_pdf.py

```diff
@@ -397,47 +480,34 @@ def get_response(
 
     elapsed_time = end_time - start_time
 
-    answer = result["result"]
-    app.chat_history += [(query, answer)]
-
-    # Prepare reranking feedback
-    rerank_feedback = ""
-    if rerank_info:
-        original_order = rerank_info["original_order"]
-        reranked_order = rerank_info["reranked_order"]
-        reranked_scores = rerank_info["reranked_scores"]
+    answer = result  # The result is now directly the answer string
 
-        # Check if the order changed
-        order_changed = original_order != reranked_order
-
-        if order_changed:
-            rerank_feedback = (
-                f"ReRanking changed document order. Top score: {reranked_scores[0]:.4f}"
-            )
-        else:
-            rerank_feedback = "ReRanking did not change document order."
+    if app.use_chat_history and app.chat_history is not None:
+        app.chat_history.add_user_message(query)
+        app.chat_history.add_ai_message(answer)
+        print(f"DEBUG: Added to chat history. Current length: {len(app.chat_history.messages)}")
+        print(f"DEBUG: Last message in history: {app.chat_history.messages[-1].content[:50]}...")
+    else:
+        print("DEBUG: Chat history not updated (disabled or None)")
 
-    # Prepare the initial output without RAGAS evaluation
+    # Prepare the output
     if is_cache_hit:
-        initial_output = f"⏱️ | Cache: {elapsed_time:.2f} SEC | COST $0.00 \n\n{rerank_feedback}\n\nRAGAS Evaluation: In progress..."
+        output = f"⏱️ | Cache: {elapsed_time:.2f} SEC | COST $0.00"
     else:
         tokens_per_sec = num_tokens / elapsed_time if elapsed_time > 0 else 0
-        initial_output = f"⏱️ | LLM: {elapsed_time:.2f} SEC | {tokens_per_sec:.2f} TOKENS/SEC | {num_tokens} TOKENS | COST ${total_cost:.4f}\n\n{rerank_feedback}\n\nRAGAS Evaluation: In progress..."
+        output = f"⏱️ | LLM: {elapsed_time:.2f} SEC | {tokens_per_sec:.2f} TOKENS/SEC | {num_tokens} TOKENS | COST ${total_cost:.4f}"
 
-    # Yield the response and initial output
+    # Yield the response and output
     for char in answer:
         history[-1][-1] += char
-        yield history, "", initial_output
+        yield history, "", output
 
     # Perform RAGAS evaluation after yielding the response
-    feedback = perform_ragas_evaluation(query, result)
+    # feedback = perform_ragas_evaluation(query, {"result": answer})
+    feedback = ""
 
     # Prepare the final output with RAGAS evaluation
-    if is_cache_hit:
-        final_output = f"⏱️ | Cache: {elapsed_time:.2f} SEC | COST $0.00 \n\n{rerank_feedback}\n\n{feedback}"
-    else:
-        tokens_per_sec = num_tokens / elapsed_time if elapsed_time > 0 else 0
-        final_output = f"⏱️ | LLM: {elapsed_time:.2f} SEC | {tokens_per_sec:.2f} TOKENS/SEC | {num_tokens} TOKENS | COST ${total_cost:.4f}\n\n{rerank_feedback}\n\n{feedback}"
+    final_output = f"{output}\n\n{feedback}"
```

Comment:
That's because there is a weird bug in RAGAS that is breaking the evaluation, we get some weird asynch error so I disabled the RAGAS evaluation until they release a fix or we find a workaround. Maybe @rbs333 can take a look and see if can deal with. 
The part that looks iffy here is that the is_cache_hit was removed, I'll check the full file to make sure that I didn't Nix it by accident.

Reply:
Do you have an example of this error? Maybe we can pull it up as a parking lot after standup

Reply:
@rbs333 I will follow up with you on the RAGAS issue